File: src/gleam_os_ffi.erl
-module(gleam_os_ffi).
-export([get_env/1, insert_env/2, delete_env/1]).

get_env(Key) ->
  case os:getenv(unicode:characters_to_list(Key)) of
    false -> {error, nil};
    Value -> {ok, unicode:characters_to_binary(Value)}
  end.

insert_env(Key, Value) ->
  os:putenv(unicode:characters_to_list(Key), unicode:characters_to_list(Value)),
  nil.

delete_env(Key) ->
  os:unsetenv(unicode:characters_to_list(Key)),
  nil.

File: src/gleam_os_ffi.js
export function insert_env(key, value) {
  process.env[key] = value;
}

export function delete_env(key) {
  delete process.env[key];
}

export function get_env(key) {
  if (key in process.env) {
    return { type: "Ok", 0: process.env[key] };
  } else {
    return { type: "Error", 0: undefined };
  }
}
